packages/reakit/src/Dialog/__utils/useEventListenerOutside.ts

```diff
@@ -58,6 +58,15 @@ export function useEventListenerOutside(
         return;
       }
 
+      // Fix for https://github.com/reakit/reakit/issues/619
+      if (
+        event.type === "focus" &&
+        (target as any) === document &&
+        document.hasFocus()
+      ) {
+        return;
+      }
```

Comment:
If it's only on focus, I think the check should be actually done here: https://github.com/reakit/reakit/blob/db71535b4cdc4d7c88f7860d3a55b62202b845c8/packages/reakit/src/Dialog/__utils/useHideOnClickOutside.ts#L62

Click does a similar check: https://github.com/reakit/reakit/blob/db71535b4cdc4d7c88f7860d3a55b62202b845c8/packages/reakit/src/Dialog/__utils/useHideOnClickOutside.ts#L40-L55

Reply:
That makes a lot of sense.